core/src/main/java/com/oath/oak/OakRValueBufferImpl.java

```diff
@@ -11,115 +11,147 @@
 import java.util.ConcurrentModificationException;
 import java.util.function.Function;
 
+import static com.oath.oak.Chunk.BLOCK_ID_LENGTH_ARRAY_INDEX;
+import static com.oath.oak.Chunk.POSITION_ARRAY_INDEX;
+import static com.oath.oak.Chunk.VALUE_BLOCK_SHIFT;
+import static com.oath.oak.Chunk.VALUE_LENGTH_MASK;
+import static com.oath.oak.ValueUtils.ValueResult.*;
+
+// remove header
 public class OakRValueBufferImpl implements OakRBuffer {
+    private long valueReference;
```

Comment:
Please add a word why each field is needed. E.g. keyreference is needed in order to retry (I assume...)